internal/cloud/openstack/imds.go

```diff
@@ -218,6 +230,20 @@ func (c *imdsClient) update(ctx context.Context) error {
 	return nil
 }
 
+func (c *imdsClient) updateUserData(ctx context.Context) error {
+	resp, err := httpGet(ctx, c.client, imdsUserDataURL)
+	if err != nil {
+		return err
+	}
+	var userdataResp userDataResponse
+	if err := json.Unmarshal(resp, &userdataResp); err != nil {
+		return fmt.Errorf("unmarshalling IMDS user_data response %q: %w", string(resp), err)
```

Comment:
```suggestion
		return fmt.Errorf("unmarshalling IMDS user_data response %q: %w", resp, err)
```
nit: superfluous cast (handled by %q)